tapchannel/aux_funding.go

```diff
@@ -752,28 +752,39 @@ func (f *FundingController) anchorVPackets(fundedPkt *tapsend.FundedPsbt,
 func (f *FundingController) signAndFinalizePsbt(ctx context.Context,
 	pkt *psbt.Packet) (*wire.MsgTx, error) {
 
-	log.Infof("Signing and finalizing PSBT w/ lnd")
+	log.Infof("Signing and finalizing PSBT w/ lnd: %v", spew.Sdump(pkt))
 
-	signedPkt, err := f.cfg.ChainWallet.SignAndFinalizePsbt(ctx, pkt)
+	// By default the wallet won't try to finalize output it sees are watch
+	// only (like the asset input), so we'll have it sign ourselves first.
+	signedPkt, err := f.cfg.ChainWallet.SignPsbt(ctx, pkt)
```

Comment:
Ah, yeah, this is probably something I could've improved in the documentation of the `SignPsbt` and `FinalizePsbt` calls. The former needs to be used for any custom signing where special instructions (like a merkle root) are present in the PSBT. The latter can only sign for "normal", keyspend wallet inputs.